avocado/utils/linux_modules.py

```diff
@@ -39,17 +39,18 @@
 BUILTIN = 2
 
 
-def load_module(module_name):
+def load_module(module_name, parameters=''):
     """
     Checks if a module has already been loaded.
     :param module_name: Name of module to check
+    :param parameters: Additional parameters to load module with
     :return: True if module is loaded, False otherwise
     :rtype: Bool
     """
     if module_is_loaded(module_name):
         return False
 
-    process.system('/sbin/modprobe ' + module_name)
+    process.system('/sbin/modprobe %s %s' % (module_name, parameters))
```

Comment:
Well the 1/2/3/4 was actually the `if then then then then else` of the implementation I'd use in the ideal world. Anyway let's get back to the earth. The simplest acceptable solution for me would be:

```
if not params:
    modprobe()
elif module_loaded():
    raise RuntimeError(...)
else:
    modprobe(..., params=params)
```

eventually we can add some of the suggested steps.

I don't understand why you find the (2) difficult as the `/sys/class///parameter` is symlink to `/sys/module//parameter`.

The (3) is safer than (4) for parameters which can be changed on runtime, because you don't remove the driver (consider enabling debug on your disk driver. It shouldn't be a problem via /sys but once you rmmod the driver you can't load it as you can't access the disk)

The (5) was not suppose to check anything, basically if you ran out of ways you raise exception as it's not possible to load the module under the current circumstances.